### PR TITLE
DinoModv3: DinoDNA

### PR DESCRIPTION
**Pull request by LyleSY**
```SUMMARY: Content "DinoMod:Adds dino eggs, dino egg laying, dino hatchlings, rebalancing weights and volumes, cleanup"```

Another step towards  #25294 this creates dino eggs to become an ingredient for raptor mutagen and potentially become a project to go into the main game, plus adjusts DinoMod weights to be more realistic per #25631 . This pull includes and updates content from #26443 in its entirety

Copies bird reproduction system, creates dino eggs and hatchlings, adds dino eggs to forage group, fixes weights to match wikipedia, removes unused item chance and luminance lines, rebalances compies to attack prey and attack the player less, removes huge dinos from the sewer, removes non-dinos, makes Deinonychus a pack hunter in caves, spellchecks the 'deinonychus' faction, removes broken mapgen files, changes raptor mutagen recipe to be slightly easier and use dino eggs, velociraptor eggs, utahraptor eggs, and deinonychus eggs.

### CODE REVIEW

File: data/mods/DinoMod/dinosaur.json [hunk 1]
```diff
@@ -12,31 +12,32 @@
...
-        "melee_skill":4,
+        "melee_skill":5,
         "melee_dice":1,
-        "melee_dice_sides":1,
-        "melee_cut":0,
+        "melee_dice_sides":2,
+        "melee_cut":1,
         "dodge":4,
-        "vision_day": 50,
+	      "vision_day": 50,
```
Comment (Reclusive-reptile) [inline data/mods/DinoMod/dinosaur.json]: Bad tab, no cookie.

**Comment by LyleSY** [comment]
I don't have authority to merge this. Can someone with access make that happen?

**Comment by LyleSY** [comment]
Thank you for looking at this @cainiaowu . I run these through the JSON Editor extension in Chrome. I ran the egg file through that again and added it fresh to the Github Desktop, but that's telling me "No local changes" which makes sense. Looking at the egg file closely, all the symbols look right. My understanding is that since these are all just JSONs, any build issues in the checker shouldn't be related. This ran fine on my machine in testing. I can't get it working now since the harvest update from yesterday, but this should merge correctly with those changes.

**Comment by KorGgenT** [comment]
> Thank you for looking at this @cainiaowu . I run these through the JSON Editor extension in Chrome. I ran the egg file through that again and added it fresh to the Github Desktop, but that's telling me "No local changes" which makes sense. Looking at the egg file closely, all the symbols look right. My understanding is that since these are all just JSONs, any build issues in the checker shouldn't be related. This ran fine on my machine in testing. I can't get it working now since the harvest update from yesterday, but this should merge correctly with those changes.

http://dev.narc.ro/cataclysm/format.html
This is the web linter spoken about. It will format your JSON to comply with the standards of this project.

**Comment by ZhilkinSerg** [comment]
Need major conflict resolution.

Marking as Stalled to clean up PR queue. Please ping me to get it reopened when you feel you can continue working on it.

**Comment by LyleSY** [comment]
Hello, can you reopen this? I practiced some with the lint cleaner and that much I can handle.

File: data/mods/DinoMod/egg.json [hunk 2]
```diff
@@ -1,4 +1,4 @@
-[
+
```
Comment (AMurkin) [inline data/mods/DinoMod/egg.json]: That was a wrong move.
```suggestion
[
```

**Comment by damien** [comment]
@LyleSY: saw this referenced in #37490—looks like you’ve done a fair bit of work already! Let me know if you’d like a hang with this, happy to pair on this with you over Discord if all you need is help linking/validating the new JSON.

Discord: `Damien#1337`

**Comment by LyleSY** [comment]
Hold on to your butts, I think I fixed all the conflicts.

**Comment by LyleSY** [comment]
Ha, merging conflicts killed some of the linting work. Relinted, trying again.

**Comment by LyleSY** [comment]
Still more linting. I feel happy.

**Comment by LyleSY** [comment]
Took me a long while to realize that the Linter automatically adds a new line at the bottom that breaks your code. Good to know!

**Comment by LyleSY** [comment]
Removed all special_freq and most special_attack code from dinosaur.json since they were deprecated in like 2014.

**Comment by damien** [comment]
Question: Does the `0.E Content Freeze` tag mean that we still have a chance of getting this into that release? @LyleSY has done a lot of work here expanding and modernizing the mod and I'd love to see it on Stable if we can swing it.

Let us know if there's anything we can do to make that happen!

**Comment by KorGgenT** [comment]
no unfortunately content freeze means it just missed the boat.

**Comment by damien** [comment]
Rats! Well, all aboard the 0.F hype train!

**Comment by LyleSY** [comment]
Playtesting revealed pretty quickly that all dino spawns were broken without debugging them in. Fixed that and tested working.

**Comment by ZhilkinSerg** [comment]
> 
> 
> Question: Does the `0.E Content Freeze` tag mean that we still have a chance of getting this into that release? @LyleSY has done a lot of work here expanding and modernizing the mod and I'd love to see it on Stable if we can swing it.
> 
> Let us know if there's anything we can do to make that happen!

Don't worry - post-`0.E` would be a quick merge train.

**Pull request merged**